src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -2078,7 +2078,7 @@ internal static BinaryOperationKind DeriveBinaryOperationKind(UnaryOperationKind
                     return BinaryOperationKind.DynamicSubtract;
 
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(incrementKind);
+                    return BinaryOperationKind.None;
```

Comment:
Hmmm.... Since this value will actually occur, maybe it should be renamed "Invalid" or some such.